public/tabs/chat/messages/message_content.tsx

```diff
@@ -37,18 +37,15 @@ export const MessageContent: React.FC<MessageContentProps> = React.memo((props)
         </div>
       );
 
-    case 'ppl_visualization': {
-      const render = chatContext.contentRenderers[props.message.contentType];
-      if (!render) return null;
-      return (
-        <div className="llm-chat-visualizations">{render({ query: props.message.content })}</div>
-      );
-    }
-
     // content types registered by plugins unknown to assistant
     default: {
       const message = props.message as IMessage;
-      return chatContext.contentRenderers[message.contentType]?.(message.content) ?? null;
+      return (
+        chatContext.contentRenderers[message.contentType]?.(message, {
```

Comment:
Shall we pass `message.content` here?

Reply:
I wandered around what we should pass here for quite a long time and finally decide to pass message here instead of `message.content` because I think `message` contains all the infos a contentType needs to render, and for message_content and customized render function, they should have the same context to render itself, instead of only a field `message.content`. It is a good chance to change what we will pass here as data grid is the first customized contentType.

Reply:
Shall we refactor `contentRenderers` to `messageRenderers`? Or we need to update the argument type at `ContentRenderer` in `public/types.ts`. We may change `content: unknown` to `message: IOutput`.

Reply:
Agree, but what about `message: IMessage`?